### ISSUE TITLE
Question: Secure Appsync API

### ISSUE THREAD

**Issue by onerider** [aws/aws-appsync-community#83]
Hi, this is more of a question than an issue but I can't figure i out how to prevent the Appsync Graphql-api from having unrestricted access no matter which client being used.
This is my scenario:
I have a React-client accessing the Appsync graphql API with some queries and mutations. The user is authenticated using Cognito. Running the app works fine, the user can query and update data as expected.
However, the very same user can access the API using a 3rd party app like Postman and send unwanted values to a mutation. These values should be calculated internally by the client application only.
How do I secure this? I.e I only want an authenticated user to be able to access the API using the client.
Thanks in advance

**Comment by sw-yx** [comment]
think this is addressed by persisted queries. i'm interested in this too.

**Comment by T-Caines** [comment]
I agree with @sw-yx, this is a use case for persisted queries. However, AppSync does not currently support this feature.
Thanks for showing your interest in this feature, I will bring forward this interest to the team in order to help get this feature prioritized in the future.

**Comment by naingaungphyo** [comment]
I also want to know how to prevent users from using their modified custom query to the appsync graphql api.
Can we block access coming from non-app 3rd party clients? (using something like IAM?)

**Comment by naingaungphyo** [comment]
I think appsync can make this happens easily because there is a resolver layer in appsync. I have an idea that it may work by compiling the query used in the app (should save in query.graphql file as an example) and generate hash code when graphql-compile and save them in the resolver template and check the hash code for every request.

Sorry to annoy and tag you. How do you think about this feature? @tinnou  @jpignata
